fir.lic

```diff
@@ -0,0 +1,171 @@
+custom_require.call(%w[common common-crafting common-items common-money common-travel drinfomon equipmanager])
+
+class FirQuest
+  include DRC
+  include DRCI
+  include DRCM
+  include DRCT
+  include DRCC
+
+  def initialize
+
+    arg_definitions = [
+      [
+        { name: 'number', regex: /\d+/, description: 'Number of Fir Familiars to make (1-3)' },
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+    EquipmentManager.new.empty_hands
+
+    settings = get_settings
+    @bag = settings.crafting_container
+    @bag_items = settings.crafting_items_in_container
+    @belt = settings.engineering_belt
+
+    unless args.number
+      num = 1
+    else
+      if args.number.to_i > 3
+        echo '****Cannot get more than 3 at a time!****'
+        exit
+      else
+        num = args.number.to_i
+      end
+    end
+
+    check_carving_knife
+    get_holy_water
+    get_fir_bark(num)
+    goto_shard_temple
+
+    unless goto_old_man
+      open_wall
+      goto_old_man
+    end
+
+    get_talisman(num)
+    move 'out'
+    walk_to 2807
+    carve_talisman(num)
+  end
+  
+    def open_wall
+    bput('put right hand in first hole','You have passed the test of lore')
+    bput('get fir bark','you get')
+    bput('put left hand in second hole','You have passed the test of the tree')
+    bput('stow right','You put')
+    bput('get my silver vial','You get')
+    bput('get water from vial','You get some holy water')
+    bput('stow vial','You put')
+    bput('put right hand in third hole','A foreboding wall moves out of your way to allow passage!')
+    bput('get my silver vial','You get')
+    bput('put water in my vial','You put')
+    bput('stow vial','You put')
+  end
+
+  def goto_old_man
+    return false if ['A voice whispers in your head',"You can't"].include? bput('west','A voice whispers in your head',"You can't",'Obvious exits')
+    move "go stairs"
+    return true
+  end
+
+  def carve_talisman(number)
+    number.times do |x|
+      ord = $ORDINALS[x]
+      bput("get fir talisman",'You get')
+      bput('rub my fir talisman','You rub')
+      get_crafting_item('carving knife', @bag, @belt, @belt)
+      bput('swap', 'You move')
+      carve
+      stow_crafting_item('carving knife', @bag, @belt)
+      bput('stow talisman','You put')
+    end
+  end
+
+  def carve
+    case bput('carve my talisman','fine details','Roundtime','Round time')
+    when /fine details/
+      return
+    else
+      carve
+    end
+  end
+
+  def get_talisman(number)
+    number.times do |x|
+      return if 'I could not find' == bput('get fir bark', 'You get', 'I could not find')
+      bput('give man', 'The old man takes your bark')
+      bput("stow right", 'You put')
+    end
+  end
+
+  def check_carving_knife
+    return if 'You tap' == bput('tap my carving knife','You tap','I could not find')
```

Comment:
`common-items` has an `exists?` helper that might make this read a bit better

Reply:
done!